src/Core/System/SystemConfig/SystemConfigService.php

```diff
@@ -158,6 +168,35 @@ public function delete(string $key, ?string $salesChannel = null): void
         $this->set($key, null, $salesChannel);
     }
 
+    /**
+     * Fetches default values from bundle configuration and saves it to database.
+     *
+     * @param Bundle              $bundle  The bundle or plugin.
+     * @param InstallContext|null $context Used for plugins.
+     *
+     * @throws Exception\BundleConfigNotFoundException
+     */
+    public function savePluginConfiguration(Bundle $bundle, InstallContext $context = null): void
```

Comment:
I would just provide a parameter if the values should be overwritten or not. so it will be independent from the plugin lifecycle

Reply:
Fixed this also.